src/Spatial/Euclidean/PolyLine2D.cs

```diff
@@ -0,0 +1,224 @@
+        /// Computes the length of the polyline by summing the lengths of the individual segments
+        /// </summary>
+        /// <returns></returns>
+        private double GetPolyLineLength()
+        {
+            double length = 0;
+            for (int i = 0; i < this._points.Count - 1; ++i)
+                length += this[i].DistanceTo(this[i + 1]);
+            return length;
+        }
+
+        /// <summary>
+        /// Get the point at a fractional distance along the curve.  For instance, fraction=0.5 will return
+        /// the point halfway along the length of the polyline.
+        /// </summary>
+        /// <param name="fraction">The fractional length at which to compute the point</param>
+        /// <returns></returns>
+        public Point2D GetPointAtFractionAlongCurve(double fraction)
+        {
+            return this.GetPointAtLengthFromStart(fraction * this.Length);
+        }
+
+        /// <summary>
+        /// Get the point at a specified distance along the curve.  A negative argument will return the first point,
+        /// an argument greater than the length of the curve will return the last point.
+        /// </summary>
+        /// <param name="lengthFromStart">The distance from the first point along the curve at which to return a point</param>
+        /// <returns></returns>
+        public Point2D GetPointAtLengthFromStart(double lengthFromStart)
+        {
+            double length = this.Length;
+            if (lengthFromStart >= length)
+                return this.Last();
+            if (lengthFromStart <= 0)
+                return this.First();
+
+            double cumulativeLength = 0;
+            int i = 0;
+            while (true)
+            {
+                double nextLength = cumulativeLength + this[i].DistanceTo(this[i + 1]);
+                if (cumulativeLength <= lengthFromStart && nextLength > lengthFromStart)
+                {
+                    double leftover = lengthFromStart - cumulativeLength;
+                    Vector2D direction = this[i].VectorTo(this[i + 1]).Normalize();
+                    return this[i] + (direction * leftover);
+                }
+                else
+                {
+                    cumulativeLength = nextLength;
+                    i++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Reduce the complexity of a manifold of points represented as an IEnumerable of Point2D objects.
+        /// This algorithm goes through each point in the manifold and computes the error that would be introduced
+        /// from the original if that point were removed.  Then it removes nonadjacent points to produce a 
+        /// reduced size manifold.
+        /// </summary>
+        /// <param name="points"></param>
+        /// <param name="tolerance"></param>
+        /// <returns></returns>
+        private static IEnumerable<Point2D> ReduceComplexitySingleStep(IEnumerable<Point2D> points, double tolerance)
```

Comment:
NIT: we have followed StyleCop ordering public before private, static before instance etc.